dmscripts/notify_suppliers_of_new_fw_cq_answers.py

```diff
@@ -0,0 +1,81 @@
+from logging import Logger
+from typing import Optional
+from uuid import UUID, uuid4
+
+from dmapiclient import DataAPIClient
+
+from dmutils.email import DMNotifyClient
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+from dmutils.env_helpers import get_web_url_from_stage
+
+
+def notify_suppliers_of_new_fw_cq_answers(
+    data_api_client: DataAPIClient,
+    notify_client: DMNotifyClient,
+    notify_template_id: str,
+    framework_slug: str,
+    stage: str,
+    dry_run: bool,
+    logger: Logger,
+    run_id: Optional[UUID] = None,
+) -> int:
+    run_is_new = bool(run_id)
+    run_id = run_id or uuid4()
+    logger.info(f"{'Starting' if run_is_new else 'Resuming'} run id {{run_id}}", extra={"run_id": str(run_id)})
+
+    framework = data_api_client.get_framework(framework_slug)["frameworks"]
+
+    if framework["status"] != "open":
+        raise ValueError(f"Framework {framework_slug!r} is not open (status is {framework['status']!r})")
+
+    failure_count = 0
+
+    for supplier_framework in data_api_client.find_framework_suppliers_iter(framework_slug):
+        for user in data_api_client.find_users_iter(supplier_id=supplier_framework["supplierId"]):
+            if user["active"]:
```

Comment:
I think I am going to create a tech debt card to allow filtering users by `active` status in the API - it seems strange that we don't have this already?

Reply:
(Now done - https://trello.com/c/YTJrDGmH/1210-allow-filtering-by-active-status-in-api-listusers-view)